agents/ahv/fence_ahv.py

```diff
@@ -0,0 +1,688 @@
+        except AHVFenceAgentException as err:
+                logging.error(err)
+                fencing.fail(fencing.EC_GENERIC_ERROR)
+
+def power_cycle(client, options):
+        """
+        Power cycle a VM
+
+        Args:
+            Client(V4NutanixClient instance): V4NutanixClient instance
+            options(dict): options dictionary
+
+        Returns:
+            None
+        """
+        vmid = None
+        name = None
+        timeout = None
+
+        if "--power-timeout" in options:
+                timeout = options["--power-timeout"]
+
+        if "--uuid" in options:
+                vmid = options["--uuid"]
+        else:
+                name = options["--plug"]
+
+        if not name and not vmid:
+                logging.error("Need VM name or VM UUID to set power cycling a VM")
+                fencing.fail(fencing.EC_GENERIC_ERROR)
+
+        try:
+                client.power_cycle_vm(vm_name=name, vm_uuid=vmid, timeout=timeout)
+        except AHVFenceAgentException as err:
+                logging.error(err)
+                fencing.fail(fencing.EC_GENERIC_ERROR)
+
+def define_new_opts():
+        """
+        Define new CLI args
+        """
+        fencing.all_opt["filter"] = {
+                "getopt": ":",
+                "longopt": "filter",
+                "help": """
+                        --filter=[filter]	Filter list, list_vms actions.
+                        --filter=\"name eq 'node1-vm'\"
+                        --filter=\"startswith(name,'node')\"
+                        --filter=\"name in ('node1-vm','node-3-vm')\" """,
+                "required": "0",
+                "shortdesc": "Filter list, list_vms"
+                             "e.g: \"name eq 'node1-vm'\"",
+                "order": 2
+        }
+
+def main():
+        """
+        Main function
+        """
+        device_opt = [
+            "ipaddr",
+            "login",
+            "passwd",
+            "ssl",
+            "notls",
+            "web",
+            "port",
+            "filter",
+            "method",
+            "disable_timeout",
+            "power_timeout"
+        ]
+
+        atexit.register(fencing.atexit_handler)
+        define_new_opts()
+
+        fencing.all_opt["method"]["default"] = "onoff"
+        fencing.all_opt["disable_timeout"]["default"] = "false"
+        fencing.all_opt["power_timeout"]["default"] = str(MIN_TIMEOUT)
+        options = fencing.check_input(device_opt, fencing.process_input(device_opt))
+        docs = {}
+        docs["shortdesc"] = "Fencing agent for Nutanix AHV Cluster VMs."
+        docs["longdesc"] = """
+                            fence_ahv is a fencing agent for nodes
+                            deployed on Nutanix AHV cluster with AHV cluster
+                            being managed by Prism Central.
+                            """
+        docs["vendorurl"] = "https://www.nutanix.com"
```

Comment:
should the URL be the doc ref for the V4 API? or is it just a vendor URL?

Reply:
Just the vendor..